helix-term/src/ui/lsp.rs

```diff
@@ -66,21 +100,31 @@ impl Component for SignatureHelp {
             )]
         });
 
+        let sig = &self.signatures[self.active_signature];
         let sig_text = crate::ui::markdown::highlighted_code_block(
-            &self.signature,
+            sig.signature.as_str(),
             &self.language,
             Some(&cx.editor.theme),
             Arc::clone(&self.config_loader),
             active_param_span,
         );
 
+        if self.signatures.len() > 1 {
+            let signature_index =
+                format!("[{}/{}]", self.active_signature + 1, self.signatures.len());
+            let sig_index = Markdown::new(signature_index, self.config_loader.clone());
+            let sig_index = sig_index.parse(Some(&cx.editor.theme));
+            let sig_index_para = Paragraph::new(&sig_index);
+            sig_index_para.render(area.with_height(1).clip_left(1), surface);
+        }
```

Comment:
Could we put this on the same line as the signature itself, right-aligned? The display being right above the signature right now seems very compact. Also minor nit but I think I would prefer regular parens here: `(1/2)` rather than `[1/2]`

Reply:
I think this should be fairly straightforward because of how we use Paragraph already here. See how diagnostics are rendered right-aligned for reference: https://github.com/helix-editor/helix/blob/7775b35cba1b09e81c90835d5755bf9144195dcb/helix-term/src/ui/editor.rs#L751-L760

Also I think we should use `Text::from` on the indicator here instead of trying to parse it as markdown

Reply:
done as you requested. only one remark. somehow I liked it better, if the signature index is up in the free line. If we put it on the same line, then we need to calculate the needed space also in `required_size`. (but this is not an issue.)

the following are the image of the current version

![sig-small](https://github.com/helix-editor/helix/assets/15311024/3d88c5af-92b3-4964-9e55-1abd8446e2c8)

![sig1](https://github.com/helix-editor/helix/assets/15311024/cb52f8da-b49c-4f10-8f4d-aac580bd8767)